ts/metrics/caching_metric.py

```diff
@@ -102,7 +102,7 @@ def emit_metrics(
         """
         metric_str = (
             f"[METRICS]{self.metric_name}.{self.unit}:{value}|#{dimension_string}|"
-            f"#hostname:{socket.gethostname()},{int(time.time())}"
+            f"#type:{self.metric_type.name}|#hostname:{socket.gethostname()},{int(time.time())}"
```

Comment:
can you add test to check the type is included in the log from <v0.6.0 api?

Reply:
Metric type has not been logged by the backend in the past:

v0.6.0
https://github.com/pytorch/serve/blob/4236a86dc0a018198ecd3fe261e835b416df739e/ts/metrics/metric.py#L69-L77

v0.8.0
https://github.com/pytorch/serve/blob/35fb57438024a6c1850e45e28b227f38617de02d/ts/metrics/caching_metric.py#L100-L105

v0.9.0
https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/caching_metric.py#L103-L110

I'm including the metric type in the backend log in this PR to support metric type detection during auto detection of metrics by the frontend.

Currently, the python API to update metrics in the backend all support a metric type argument, with default being the `counter` type. Also these APIs are backwards compatible with `v0.6.0`.

https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/metric_cache_abstract.py#L106-L130
https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/metric_cache_abstract.py#L73-L81
https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/metric_cache_abstract.py#L106-L112
https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/metric_cache_abstract.py#L132-L140
https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/metric_cache_abstract.py#L169-L177
https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/metric_cache_abstract.py#L206-L213
https://github.com/pytorch/serve/blob/db47936156a3e5803b42b38001f0f43affb7c720/ts/metrics/metric_cache_abstract.py#L236-L257

Since metric type is set by all of the APIs above, we can emit the metric type in the backend and detect the same in the frontend.

Reply:
[Integration tests](https://github.com/pytorch/serve/pull/2769/files#diff-a3a69361508ac6187f63a93ee979c96c5ff0e792674c77f96f2a4a5b56158fbeR552-R654) for metrics auto detection which also includes metrics type detection in the frontend.